api_tests/nodes/views/test_node_list.py

```diff
@@ -96,6 +97,35 @@ def test_omit_retracted_registration(self):
         res = self.app.get(self.url, auth=self.user.auth)
         assert_equal(len(res.json['data']), 2)
 
+    def test_node_list_has_root(self):
+        res = self.app.get(self.url, auth=self.user.auth)
+        projects_with_root = 0
+        for project in res.json['data']:
+            if project['relationships'].get('root', None):
+                projects_with_root += 1
+        assert_equal(projects_with_root, 2)
+
+    def test_node_list_has_proper_root(self):
+        project_one = ProjectFactory(title="Project One", is_public=True)
+        ProjectFactory(parent=project_one, is_public=True)
+
+        res = self.app.get(self.url+'?embed=root&embed=parent', auth=self.user.auth)
+
+        projects_with_correct_root = 0
+        for project in res.json['data']:
+            root_id = project['embeds']['root']['data']['id']
+
+            if project['embeds']['parent'].get('data'):
```

Comment:
There's a lot of logic here, which makes this test hard to follow. If you're just trying to check that the root IDs are correct, then you can simplify. How about something along the lines of:

```python
for project_json in res.json['data']:
     # Load the project from the DB
    project = Node.load(project_json['id'])
    # The root ID in the JSON should be the correct root ID
    assert_equal(project_json['embeds']['root']['data']['id'], project.root._id)
```